### PR TITLE
feat：泛型cache替换

### PR DESCRIPTION
**Pull request by LidolLxf**
feat：泛型cache替换

### CODE REVIEW

File: bcs-services/bcs-webconsole/console/components/bcs/project_test.go [hunk 1]
```diff
@@ -29,3 +33,44 @@ func TestGetProject(t *testing.T) {
...
+		CcBizID:       "1",
+		Creator:       "joker",
+		Kind:          "project",
+		RawCreateTime: "20240220",
+	}
+	goCache := cachenum30.New[*Project](5*time.Second, 10*time.Second)
+	for i := 0; i < b.N; i++ {
+		goCache.Set(cacheKey, &project, patrickmn.DefaultExpiration)
+		if cacheResult, ok := goCache.Get(cacheKey); ok {
+			assert.Equal(b, cacheResult, &project)
```
Comment (ifooth) [inline bcs-services/bcs-webconsole/console/components/bcs/project_test.go]: 压测这里优化下
- 正常是set后多次get, 这里 set / get 分2个函数单独压测下
- 压测一般不需要assert吧，正确性用单元测试就好，这里可以去掉
- 测试后截图发到评论，BenchmarkPatrickmn 这个可以去掉(github.com/patrickmn/go-cache就可以不用导入)

File: bcs-services/bcs-webconsole/console/components/bcs/project.go [hunk 2]
```diff
@@ -33,6 +33,9 @@ type Project struct {
 	RawCreateTime string `json:"createTime"`
 }
 
+// localProjectCache :
```
Comment (ifooth) [inline bcs-services/bcs-webconsole/console/components/bcs/project.go]: 注释: project cache with generics

Comment (LidolLxf) [reply bcs-services/bcs-webconsole/console/components/bcs/project.go]: 测试结果如图

Comment (LidolLxf) [reply bcs-services/bcs-webconsole/console/components/bcs/project.go]: $ go test -benchmem -bench=. ./...
goos: windows
goarch: amd64
pkg: github.com/Tencent/bk-bcs/bcs-services/bcs-webconsole/console/components/bcs
cpu: Intel(R) Core(TM) i7-10700 CPU @ 2.90GHz
BenchmarkPatrickmnSet-16        31682498                38.26 ns/op            0 B/op          0 allocs/op
BenchmarkPatrickmnGet-16        59813083                19.68 ns/op            0 B/op          0 allocs/op
BenchmarkCacheNum30Set-16       31740069                37.88 ns/op            0 B/op          0 allocs/op
BenchmarkCacheNum30Get-16       64346958                19.39 ns/op            0 B/op          0 allocs/op
PASS
ok      github.com/Tencent/bk-bcs/bcs-services/bcs-webconsole/console/components/bcs    5.417s

Comment (LidolLxf) [reply bcs-services/bcs-webconsole/console/components/bcs/project.go]: 微信图片_20240223113214

**Review by ifooth — APPROVED** [review]

**Pull request merged**